src/components/utils/src/threads/thread_qt.cc

```diff
@@ -0,0 +1,223 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <errno.h>
+#include <limits.h>
+#include <stddef.h>
+#include <signal.h>
+
+#include "utils/threads/thread.h"
+#include "qthread.h"
+#include "utils/atomic.h"
+#include "utils/threads/thread_delegate.h"
+#include "utils/logger.h"
+
+#include <QtConcurrent/QtConcurrent>
+
+namespace threads {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "Utils")
+
+void sleep(uint32_t ms) { QThread::msleep(ms); }
+
+size_t Thread::kMinStackSize = 0; /* Ubuntu : 16384 ; QNX : 256; */
+
+void Thread::cleanup(void* arg) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  Thread* thread = reinterpret_cast<Thread*>(arg);
+  sync_primitives::AutoLock auto_lock(thread->state_lock_);
+  thread->isThreadRunning_ = false;
+  thread->state_cond_.Broadcast();
+}
+
+void* Thread::threadFunc(void* arg) {
+  LOG4CXX_DEBUG(logger_,
+                "Thread #" << QThread::currentThread()
+                           << " started successfully");
+
+  threads::Thread* thread = reinterpret_cast<Thread*>(arg);
+  DCHECK(thread);
+
+  thread->state_lock_.Acquire();
+  thread->state_cond_.Broadcast();
+
+  while (!thread->finalized_) {
+    LOG4CXX_DEBUG(logger_,
+                  "Thread #" << QThread::currentThreadId() << " iteration");
+    thread->run_cond_.Wait(thread->state_lock_);
+    LOG4CXX_DEBUG(logger_,
+                  "Thread #" << QThread::currentThreadId() << " execute. "
+                             << "stopped_ = "
+                             << thread->stopped_
+                             << "; finalized_ = "
+                             << thread->finalized_);
+    if (!thread->stopped_ && !thread->finalized_) {
+      thread->isThreadRunning_ = true;
+      thread->state_lock_.Release();
+      thread->delegate_->threadMain();
+      thread->state_lock_.Acquire();
+      thread->isThreadRunning_ = false;
+    }
+    thread->state_cond_.Broadcast();
+    LOG4CXX_DEBUG(logger_,
+                  "Thread #" << QThread::currentThreadId()
+                             << " finished iteration");
+  }
+
+  thread->state_lock_.Release();
+  LOG4CXX_DEBUG(logger_,
+                "Thread #" << QThread::currentThreadId()
+                           << " exited successfully");
+  return NULL;
+}
+
+void Thread::SetNameForId(const PlatformThreadHandle& thread_id,
+                          std::string name) {}
```

Comment:
This function is present for compatibility with other platforms.
Such changes must be done within the overall refactoring

________________________________

This e-mail and any attachment(s) are intended only for the recipient(s) named above and others who have been specifically authorized to receive them. They may contain confidential information. If you are not the intended recipient, please do not read this email or its attachment(s). Furthermore, you are hereby notified that any dissemination, distribution or copying of this e-mail and any attachment(s) is strictly prohibited. If you have received this e-mail in error, please immediately notify the sender by replying to this e-mail and then delete this e-mail and any attachment(s) or copies thereof from your system. Thank you.